doc/design/design_model_zoo_pai.md

```diff
@@ -0,0 +1,97 @@
+```bash
+$ odpscmd -e "wait instanceID"
+```
+to attach to the running task.
+
+
+2. Use the MaxCompute Python Package `pyodps`:
+```python
+
+from odps import ODPS
+
+conn = ODPS(...)
+inst = conn.run_xflow('tensorflow', parameters={'script':      'file:///my/training/script.py',
+               	                                'other_param': 'other_param_value' ... })
+print(instance.get_logview_address())
+```
+This code snippet asynchronously submits a task to PAI and prints the logview URL to stdout.
+
+### Train/Predict/Analyze in SQLFlow with Model Zoo
+In SQLFlow, we train a model with the following SQLFlow extened SELECT statement:
+```sql
+SELECT * FROM train_table ... TO TRAIN MyAwesomeEstimator ... INTO my_awesome_model;
+```
+and predict with:
+```sql
+SELECT * FROM pred_table ... TO PREDICT using my_awesome_model ... ;
+```
+and analyze with:
+```sql
+SELECT * FROM train_table ...  ANALYZE my_awesome_model USING TreeExplainer;
+
+```
+The extended SELECT statements are executed as described in [Model Zoo Design Doc](./design_model_zoo.md#Submitter-Programs):
+- If the model **is not** from a model zoo, The SQLFlow server converts these statements to a *submitter* file and submits the file to a specified engine.
+- If the model **is** from a model zoo, the submmiter program is mapped into a Docker image. the SQLFlow server calls k8s API to launch the Docker image on a k8s cluster.
+
+## The Design
+### Base Image
+The base image of the SQLFlow model zoo should incorporate both odpscmd(which is already in place) and pyodps.  We propose to use `odpscmd` at the moment because it requires less code modification in both python and golang.
+
+### Submit a SQLFlow Task to PAI With Model Zoo
+When a user submits a SELECT statement as above, SQLFlow should take the following actions:
+1. SQLFlow Checks whether the entity after `TO TRAIN/PREDICT/ANALYZE` is from a SQLFlow model zoo. For example, a plain `DNNClassier` implies that the model is a premade estimator, and `"models.sqlflow.org/sqlflow/my_awesome_model"` implies that the model is from model zoo "models.sqlflow.org". The actual mechanism may be more complicated and is still under progress.
+1. Case A, the model **is not** from a model zoo:
+    1. Case A.1: the user specifies that the task should be submitted to PAI (For example, the user set a environment variable such as `SQLFLOW_submitter=PAI`).
+        - The SQLFlow server generates a submitter script with PAI-required command line options.
+		- The SQLFlow server uses odpscmd or pyodps to submit the script to PAI as [described above](#Submit-a-Task-to-PAI). 
+    1. Case A.2: the task is not to be submitted to PAI:
+	    - Run the other way.
+1. Case B: the model **is** from a model zoo:
+    1. Case B.1: the user specifies that the task should be submitted to PAI.
```

Comment:
Currently, each deployment have been configured to use one submitter. The user can not specify which submitter to use.

Reply:
Done.

> Currently, each deployment have been configured to use one submitter. The user can not specify which submitter to use.